<div align="center"><b><strong>

LAPORAN PRAKTIKUM

ALGORITMA DAN PEMOGRAMAN

“TIPE DATA DAN VARIABLE"

Disusun Oleh :

Aria Ramadhan

2611532007

Dosen Pengampu:

Wahyudi.DR.. S.T.M.T

Asisten Praktikum:

Rifki Yuliandra


<img src="Images/unand.png" width="200">

<br>
FAKULTAS TEKNOLOGI INFORMASI

DEPARTEMEN INFORMATIKA

UNIVERSITAS ANDALAS

2026

</strong></b></div>

<h1 align = "center">

<b>KATA PENGANTAR</b>
</h1>

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkatnya, sehingga penulis dapat menyelesaikan laporan praktikum tipe data dan variable yang telah dilaksanakan sebagai pertemuan ke-2 untuk mata kuliah Praktikum Algoritma dan Pemograman. Praktikum ini berfokus pada pembahasan macam macam tipe data,serta penggunaan variable.

Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami dan dapat menggunakan berbagai tipe data.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

Padang, 2026

---
<b><h1 align="center"> 

BAB I<br>
PENDAHULUAN
</h1></b>


## 1.1 Latar Belakang
Di antara banyaknya bahasa pemrograman yang ada, Python menjadi salah satu yang paling populer dan banyak digunakan. Alasan utamanya adalah karena Python memiliki sintaksis yang sederhana, mudah dibaca seperti bahasa manusia, dan sangat fleksibel untuk berbagai kebutuhan pengembangan perangkat lunak.
Saat menulis sebuah program komputer, aktivitas utama yang kita lakukan sebenarnya adalah mengolah informasi atau data. Agar data tersebut bisa diolah oleh komputer, kita memerlukan dua komponen paling mendasar dalam pemrograman, yaitu variabel dan tipe data.
Variabel dapat dianalogikan sebagai sebuah wadah atau boks penyimpanan di dalam memori komputer.tipe data berperan sebagai label yang memperjelas jenis nilai yang disimpan—apakah itu berupa angka bulat (Integer), angka desimal (Float), teks (String), atau logika benar/salah (Boolean).
praktikum ini penting untuk dilakukan agar kita tidak hanya memahami teorinya di kelas, melainkan dapat mempraktikkan langsung cara membuat variabel yang benar, melakukan konversi (casting), serta memanipulasi berbagai tipe data secara efektif dalam Python.

## 1.2 Tujuan
Tujuan dari pelaksanaan praktikum dan penyusunan laporan ini adalah:
1. Memahami konsep dasar variabel sebagai wadah penyimpanan data di memori serta aturan penulisan nama variabel (naming convention) yang benar di Python.
2. Mengenali dan membedakan berbagai jenis tipe data dasar (seperti Integer, Float, String, Boolean) dan tipe data koleksi yang digunakan dalam Python.
3. Mampu memanipulasi variabel dan tipe data, termasuk melakukan proses konversi (type casting) secara tepat agar tidak terjadi error saat program dijalankan.
4. Melatih logika berpikir dalam memilih tipe data dan struktur variabel yang paling efisien untuk menyelesaikan suatu masalah pemrograman.
## 1.3 Manfaat
1. Membangun fondasi coding yang kuat.
2. Meningkatkan kemampuan analisis dan debugging.
3. Membentuk kebiasaan menulis kode yang rapi

---
<b><h1 align="center"> 

BAB II<br>PEMBAHASAN
</h1></b>


## 2.1 Dasar Teori

### 2.1.1 Tipe Data dan Variable
#### 2.1.1.1 Definisi 

 Variable dalam python adalah nama yang digunakan untuk menetapkan sebuah nilai atau objek. Secara teknis variable adalah referensi atau label yang menunjuk ke suatu nilai dalam memori komputer. Variable digunakan untuk menyimpan data yang dapat diakses dan diubah selama program berjalan. Tipe data dalam Python adalah jenis nilai yang dapat ditampung oleh sebuah variabel untuk diproses oleh program.

 Dalam python,anda tidak perlu mendeklarasikan tipe data secara eksplisit seperti beberapa bahasa pemrograman lain.Python menggunakan konsep "Dynamic typing",yang berarti tipe data variable ditentukan secara otomatis sesuai nilai yang diberikan.

#### 2.1.1.2 Macam - macam Tipe Data dan Variable Dalam Python

Di Python,terdapat beberapa jenis variable yang bisa digunakan. Beberapa jenis variable yang umum di gunakan di python antara lain

1. Variable string<br>Variable String adalah variable yang digunakan untuk menyimpan teks. String dalam python dapat ditulis dalam tanda kutip tunggal(''), tanda kutip ganda("") dan tanda kutip tiga("'"' atau '"'"). Variable juga mendukung operasi penggambungan (concatenatian), pengindeksan (indexing) dan sclising. Variable ini bersifat imutable artinya setelah string dibuat tidak dapat dirubah tapi dapat membuat string baru dari string yang sudah ada.

2. Variable Interger<br>Variable Interger adalah variable yang diguakan untuk menyimpan bilangan bulat(tanpa desimal). Variable ini dapat dibuat dengan menetapkan nilai langsung atau  dengan operasi matematika. Variable integer dalam python
memiliki batas atas atas dan batas bawah, umumnya batas atasnya 2 miliar untuk integer signed (bilangan bulat dengan tanda) dan 4 miliar untuk integer unsigned (bilangan bulat tanpa tanda), namun batasnya berbeda tergantung versi python yang digunakan.

3. Variable float<br>Variable Float adalah variable yang digunakan untuk menyimpan bilangan pecahan atau desimal. Walaupun kemampuannya dalam menyimpan bilangan desimal dalam presisi tinggi, namun meiliki keterbatasan dalam reprentasi angka desimal tertentu yang menyebabkan beberapa operasi matematika menghasilkan nilai tidak tepat.

4. Boolean<br>Variable Boolean adalah variable yang memiliki 2 nilai: True atau False. Variable ini digunakan untuk menyimpan nilai kebenaran yang sering di gunakan dalam sistem alur program seperti kondisional(if,elif dan else) dan loop(while dan for).




## 2.2 Kode Program

### 2.2.1 String_2611532007
Jalankan sel kode di bawah ini agar Jupyter otomatis menyimpan kodenya beserta outputnya sebagai satu kesatuan (tidak perlu screenshot terpisah untuk hasil run).

In [1]:
# Buat nama file dengan nama String_2611532007.py
#nama variable ditambah 4 digit nim terakhir contoh: nama_1234
#ganti programnya dan data masing2
nama_2007 ="Aria"
jenis_kelamin_2007 = 'L'
alamat_2007 = """
    Asrama Unand
    kec.pauh
    kota padang
"""
agama_2007 = 'islam'
print("Nama saya adalah: ",nama_2007)
print("Jenis kelamin saya adalah: ",jenis_kelamin_2007)
print("Alamat saya adalah: ", alamat_2007)
print("Agama saya adalah: ", agama_2007)


Nama saya adalah:  Aria
Jenis kelamin saya adalah:  L
Alamat saya adalah:  
    Asrama Unand
    kec.pauh
    kota padang

Agama saya adalah:  islam


**Penjelasan :**
1. Baris 1-3 adalah baris komentar ini cuma catatan/keterangan buat programer, bukan perintah.(#) untuk komentar 1 baris, sedangkan ("""...""") untuk komentar banyak baris.
2. Baris 4 membuat variabel bernama nama_2007 dan mengisinya dengan teks/string "Aria".
3. Baris 5 membuat variabel jenis_kelamin_2007 dan mengisinya dengan karakter 'L'
4. Baris 6 membuat variabel alamat_2007 menggunakan tiga tanda petik ("""). Tiga tanda petik digunakan untuk membuat string banyak baris (multiline string), sehingga format baris baru dan spasi di dalamnya tetap terjaga.
5. Baris 7 membuat variabel agama_2007 dan mengisinya dengan teks/string 'islam'.
6. Baris 8 mencetak kalimat "Nama saya adalah: " diikuti oleh nilai dari variabel nama_2007 (yaitu "Aria").
7. Baris 9 mencetak kalimat "Jenis kelamin saya adalah: " diikuti oleh nilai dari variabel jenis_kelamin_2007 (yaitu 'L').
8. Baris 10 mencetak kalimat "Alamat saya adalah: " diikuti oleh isi variabel alamat_2007 yang mencakup teks multiline.
9. Baris 11 mencetak kalimat "Agama saya adalah: " diikuti oleh nilai dari variabel agama_2007 (yaitu 'islam').

### 2.2.2 Numerik1_2611532007
Jalankan sel kode di bawah ini agar Jupyter otomatis menyimpan kodenya beserta outputnya sebagai satu kesatuan (tidak perlu screenshot terpisah untuk hasil run).

In [2]:
#create a variable with integer value
a_2007=100
print("the type of variable having value",a_2007,"is",type(a_2007))

#create a variable with float value
b_2007=10.2345
print("the type of variable having value",b_2007,"is",type(b_2007))

#create a variable with complex value
c_2007=100+3j
print("the type of variable having value",c_2007,"is",type(c_2007))


the type of variable having value 100 is <class 'int'>
the type of variable having value 10.2345 is <class 'float'>
the type of variable having value (100+3j) is <class 'complex'>


**Penjelasan :**
1. Baris 1,5 dan 9 adalah komentar (comment)
2. Baris 2 Membuat variabel bernama a_2007 dan mengisinya dengan nilai 100 (tipe data int).
3. Baris 3 Mencetak teks, nilai dari a_2007 (100), dan tipe datanya yang diperiksa menggunakan fungsi type(a_2007) (menghasilkan <class 'int'>).
4. Baris 6 Membuat variabel b_2007 dan mengisinya dengan nilai 10.2345 (tipe data float).
5. Baris 7 Mencetak teks, nilai dari b_2007 (10.2345), dan tipe datanya via type(b_2007) (menghasilkan <class 'float'>).
6. Baris 10 Membuat variabel c_2007 dan mengisinya dengan nilai 100+3j (tipe data complex, di mana 100 adalah bagian riil dan 3j adalah bagian imajiner).
7. Baris 11 Mencetak teks, nilai dari c_2007 ((100+3j)), dan tipe datanya via type(c_2007) (menghasilkan <class 'complex'>).

### 2.2.3 Boolean_2611532007
Jalankan sel kode di bawah ini agar Jupyter otomatis menyimpan kodenya beserta outputnya sebagai satu kesatuan (tidak perlu screenshot terpisah untuk hasil run).

In [3]:
# Buat file dengan nama Boolean_2611532007.py
# Nama variable ditambah 4 digit nim terakhir contoh: nilai_1234
# Deklarasi variable dengan tipe data Boolean
is_lulus_2007 = True
is_cumlaude_2007 = True

# Menggunakan Boolean
nilai_2007 = 85
batas_lulus_2007 = 75

# Menentukan nilai Boolean dari kondisi
status_kelulusan_2007 = nilai_2007 >= batas_lulus_2007  # Hasilnya akan True

print("=== Check Kelulusan ===")
print("Nilai:", nilai_2007)
print("Apakah Lulus?:", status_kelulusan_2007)
if is_lulus_2007 and is_cumlaude_2007 :
    print("Selamat, anda lulus dengan predikat Cumlaude")

=== Check Kelulusan ===
Nilai: 85
Apakah Lulus?: True
Selamat, anda lulus dengan predikat Cumlaude


**Penjelasan :**
1. Baris 1-3,7 dan 11 merupakan baris yang komentar
2. Baris 4 Membuat variabel is_lulus_2007 bernilai True (benar).
3. Baris 5 Membuat variabel is_cumlaude_2007 bernilai True (benar).
4. Baris 8 Membuat variabel nilai_2007 bernilai 85 (tipe integer).
5. Baris 9 Membuat variabel batas_lulus_2007 bernilai 75 (tipe integer).
7. Baris 12 Mengevaluasi apakah 85 >= 75. Karena nilainya benar, variabel status_kelulusan_2007 menyimpan nilai Boolean True.
8. Baris 14 Mencetak teks judul "=== Check Kelulusan ===" ke layar.
9. Baris 15 Mencetak label "Nilai:" beserta isi variabel nilai_2007 (85).
10. Baris 16 Mencetak label "Apakah Lulus?:" beserta hasil evaluasi variabel status_kelulusan_2007 (True).
11. Baris 17 Pengondisian menggunakan operator logika and. Memeriksa apakah **kedua** variabel (is_lulus_2007 DAN is_cumlaude_2007) bernilai True.
12. Baris 18 ini akan dieksekusi dan mencetak teks pesan selamat jika kondisi pada perintah if terpenuhi.

### 2.2.4 Input_2611532007
Jalankan sel kode di bawah ini agar Jupyter otomatis menyimpan kodenya beserta outputnya sebagai satu kesatuan (tidak perlu screenshot terpisah untuk hasil run).

In [4]:
# Buat file dengan nama Input_2611532007.py
# Buat program untuk menghitung luas persegi panjang
# Nama variable ditambah 4 digit nim terakhir contoh: panjang_1234
# Program ini menggunakan fungsi input() untuk meminta pengguna 
# Memasukan nilai panjang dan lebar persegi panjang.
# Nilai yang dimasukkan akan dikonversi menjadi tipe data interger
panjang_2007 = input('masukkan nilai panjang: ')
lebar_2007 = input('masukkan nilai lebar: ')
luas_2007 = int(panjang_2007) * int(lebar_2007)
print("luas =", luas_2007)

luas = 2475


**Penjelasan :**
1. Baris 1-6 merupakan baris komentar yang berisi intruksi yang akan di jalankan di program ini
2. Baris 7 Menampilkan pesan ke layar dan meminta masukan (*input*) dari pengguna untuk nilai panjang, lalu menyimpannya ke dalam variabel panjang_2007 (bertipe data string).
3. Baris 8 Menampilkan pesan ke layar dan meminta masukan dari pengguna untuk nilai lebar, lalu menyimpannya ke dalam variabel lebar_2007 (bertipe data string).
4. Baris 9 Mengubah (*type casting*) isi variabel panjang_2007 dan lebar_2007 yang tadinya teks/string menjadi bilangan bulat dengan fungsi int(), kemudian mengalikan keduanya untuk menghitung luas dan hasilnya disimpan ke variabel luas_2007.
5. Baris 10 Mencetak teks "luas =" beserta nilai hasil perhitungan yang tersimpan pada variabel luas_2007 ke layar.

### 2.2.5 Konstanta_2611532007
Jalankan sel kode di bawah ini agar Jupyter otomatis menyimpan kodenya beserta outputnya sebagai satu kesatuan (tidak perlu screenshot terpisah untuk hasil run).

In [5]:
# Buat file dengan nama Konstanta_2611532007.py
# Program ini menggunakan konstanta untuk menghitung luas lingkaran
# nama variable ditambah 4 digit nim terakhir contoh: jari_1234

from typing import Final
PI: Final = 3.14
print("pi: %f" % (PI))
jari_2007 = float(input('Masukkan nilai jari-jari: '))
luas_2007 = PI * jari_2007 * jari_2007
print("Luas lingkaran dengan jari jari %.2f adalah %.2f" % (jari_2007, luas_2007))

pi: 3.140000
Luas lingkaran dengan jari jari 12.00 adalah 452.16


**Penjelasan :**
1. Baris 1-3 merupakan baris komentar yang membarikan intruksi tentang program .
2. Baris 4 Mengimpor penanda tipe (*type hint*) Final dari modul bawaan typing untuk menegaskan bahwa suatu variabel ditujukan sebagai konstanta (nilainya tidak boleh diubah).
3. Baris 5 Mendeklarasikan konstanta bernama PI bernilai 3.14 dan menandainya menggunakan : Final agar diperlakukan sebagai nilai tetap.
4. Baris 6 Mencetak nilai konstanta PI menggunakan format penulisan string legacy (%f), yang secara default akan menampilkan 6 angka di belakang koma (3.140000).
Meminta masukan (*input*) dari pengguna untuk nilai jari-jari, lalu langsung mengonversi teks masukan tersebut menjadi tipe bilangan desimal menggunakan fungsi float(), dan menyimpannya ke variabel jari_2007.
5. Baris 7 Menghitung luas lingkaran dengan rumus $\text{PI} \times r^2$ (PI * jari_2007 * jari_2007) dan menyimpannya ke dalam variabel luas_2007.
6. Baris 8 Mencetak hasil ke layar menggunakan format penulisan string (%.2f), di mana %.2f membatasi tampilan angka desimal variabel jari_2007 dan luas_2007 agar hanya memunculkan 2 angka di belakang koma.

---
<b><h1 align = "center"></b>

**BAB III<br>KESIMPULAN**</h1>

Praktikum “Tipe Data dan Variable” memberikan pemahaman tentang berbagai variable di dalam bahasa pemrograman python dan bagaimana cara menggunakan berbagai variable tersebut sebagai langkah awal untuk memuklai program yang lebih kompleks.

---
<b><h1 align = "center">

DAFTAR PUSTAKA</h1></b>

1. Dr. Afiyati,S. Si. ,M. T. "Bahasa Pemrograman Python." Sada Kurnia Pustaka, 2024